perun-core/src/main/java/cz/metacentrum/perun/core/entry/GroupsManagerEntry.java

```diff
@@ -1132,11 +1177,17 @@ public RichGroup getRichGroupByIdWithAttributesByNames(PerunSession sess, int gr
 	}
 
 	@Override
-	public Group createGroupUnion(PerunSession sess, Group resultGroup, Group operandGroup) throws InternalErrorException, GroupNotExistsException, PrivilegeException, GroupRelationNotAllowed, GroupRelationAlreadyExists, WrongAttributeValueException, WrongReferenceAttributeValueException {
+	public Group createGroupUnion(PerunSession sess, Group resultGroup, Group operandGroup) throws InternalErrorException, GroupNotExistsException, PrivilegeException, GroupRelationNotAllowed, GroupRelationAlreadyExists, WrongAttributeValueException, WrongReferenceAttributeValueException, ExternallyManagedException {
 		Utils.checkPerunSession(sess);
 		getGroupsManagerBl().checkGroupExists(sess, resultGroup);
 		getGroupsManagerBl().checkGroupExists(sess, operandGroup);
 
+		if (getGroupsManagerBl().isGroupInStructureSynchronizationTree(sess, resultGroup)) {
+			throw new ExternallyManagedException("Result group: " + resultGroup + " is externally managed!");
+		} else if (getGroupsManagerBl().isGroupInStructureSynchronizationTree(sess, operandGroup)) {
+			throw new ExternallyManagedException("Operand group: " + operandGroup + " is externally managed!");
```

Comment:
I am not sure if this is really a problem. Because you can always take all members from synchronized group and add them to the different group by union. What do you think?